src/viam/sdk/CMakeLists.txt

```diff
@@ -237,6 +237,8 @@ target_link_libraries(viamsdk
   PRIVATE absl::strings
   PRIVATE viam_rust_utils
   PRIVATE Threads::Threads
+  PRIVATE $<$<PLATFORM_ID:Linux>:dl>
```

Comment:
I wanted to ask you about that, it comes up in the conanfile too. It seems to be idiomatic to do the following:
https://github.com/conan-io/conan-center-index/blob/master/recipes/grpc/all/conanfile.py#L302
And I figured we may as well do the same even if we don't claim in the README to support BSD. But yes I can do a `Linux || BSD` clause here and in the conanfile

Reply:
In the sdk/CML.txt we do an `if (APPLE)` to link `-framework Security` so I may just spell this as an `if (NOT APPLE)`

Reply:
Yeah, the reason for that `-framework Security` is interesting, btw: https://github.com/viamrobotics/viam-cpp-sdk/pull/146/files#r1334516668.

Essentially, it is viam_rust_utils that has a dependency on the `Security` framework on macOS. However, since we don't obtain viam_rust_utils via a find_package like thing, there is no way for it to express its transitive dependencies.

I'm not particularly happy about needing to do that, but I don't have a better answer.

Regarding your suggestion to make this just be "not apple", I think that makes sense. Maybe do "not apple and not windows?"

Reply:
ahh that explains a lot, I think that's why I ended up needing to add `librt` and `libdl` too, I was getting linker errors from `viam_rust_utils` on the static build without them.

As for not windows, I don't think there is currently any mention of windows in our build files, are we hoping to be able to offer it down the road?